### PR TITLE
Data time series class

### PR DESCRIPTION
**Pull request by mSitkovets**
### Related Issue:

None

### Proposed Changes

Adding a class named `DailyTimeSeries` to wrap the utility functions defined in `get_data.py`. The `CovidAPIv2Integrator` would have a has_a relationship with `DailyTimeSeries` and have it passed into its initializer so that it can be used to get the daily time series for all countries or specifically for the US by sending the `US=True` flag when calling the `get_data_time_series` method. In addition, this could further be refactored in the future to instead pass in the name of the country as an argument if we wish to add this functionality to the Covid19 API.

### Addtional Info

- any additional information or context

### Checklist

* [x] Tests
* [x] Translations
* [x] Documentation

### CODE REVIEW

File: app/utils/get_data.py [hunk 1]
```diff
@@ -46,42 +46,31 @@ def get_data_daily_reports(self, US: bool = False) -> pd.DataFrame:
...
+            categories = JHU_CSSE_FILE_PATHS['CATEGORIES'][:-1]
+            url = JHU_CSSE_FILE_PATHS['BASE_URL_US_TIME_SERIES']
+        # Iterate through all files
+        for category in categories:
+            url = url.format(category)
+
+            # Extract data
+            df = pd.read_csv(url)
+            df = helper_df_cleaning(df)
+            if US:
```
Comment (nat236919) [inline app/utils/get_data.py:68]: I think we can create a cleaning process as a separate private method

```
# Get data from time series (General and US)
class DataTimeSeries:
    """ Get the tiemseires dataset from JHU CSSE and Prepare DataFrames """
    def get_data_time_series(self, US: bool = False) -> Dict[str, pd.DataFrame]:
        """ Get the dataset from JHU CSSE """
        dataframes = {}

        # Determine categories and url
        if US:
            categories = JHU_CSSE_FILE_PATHS['CATEGORIES'][:-1] # Select only 'confirmed' and 'deaths'
            url = JHU_CSSE_FILE_PATHS['BASE_URL_US_TIME_SERIES']
        else:
            categories = JHU_CSSE_FILE_PATHS['CATEGORIES']
            url = JHU_CSSE_FILE_PATHS['BASE_URL_TIME_SERIES']

        # Iterate through all files
        for category in categories:
            url = url.format(category)
            # Extract data from URL
            df = pd.read_csv(url)
            df = self._clean_timeseries_dataframe(df, US)
            dataframes[category] = df

        return dataframes
    
    def _clean_timeseries_dataframe(self, df: pd.DataFrame, US: bool = False) -> pd.DataFrame:
        df_cleaned = helper_df_cleaning(df) # main pre-processing
        if US:
            df_cleaned = helper_df_cols_cleaning(df_cleaned, ['Lat', 'Long_'], float)
        return df_cleaned
```

What do you think?

**Review by nat236919 — COMMENTED** [review]
I think we can create a cleaning process as a separate private method

```
# Get data from time series (General and US)
class DataTimeSeries:
    """ Get the tiemseires dataset from JHU CSSE and Prepare DataFrames """
    def get_data_time_series(self, US: bool = False) -> Dict[str, pd.DataFrame]:
        """ Get the dataset from JHU CSSE """
        dataframes = {}

        # Determine categories and url
        if US:
            categories = JHU_CSSE_FILE_PATHS['CATEGORIES'][:-1] # Select only 'confirmed' and 'deaths'
            url = JHU_CSSE_FILE_PATHS['BASE_URL_US_TIME_SERIES']
        else:
            categories = JHU_CSSE_FILE_PATHS['CATEGORIES']
            url = JHU_CSSE_FILE_PATHS['BASE_URL_TIME_SERIES']

        # Iterate through all files
        for category in categories:
            url = url.format(category)
            # Extract data from URL
            df = pd.read_csv(url)
            df = self._clean_timeseries_dataframe(df, US)
            dataframes[category] = df

        return dataframes
    
    def _clean_timeseries_dataframe(self, df: pd.DataFrame, US: bool = False) -> pd.DataFrame:
        df_cleaned = helper_df_cleaning(df) # main pre-processing
        if US:
            df_cleaned = helper_df_cols_cleaning(df_cleaned, ['Lat', 'Long_'], float)
        return df_cleaned
```
What do you think?

**Review by nat236919 — APPROVED** [review]
LGTM

**Comment by mSitkovets** [comment]
> I think we can create a cleaning process as a separate private method

Yes that does make it cleaner!

**Comment by nat236919** [comment]
@all-contributors please add @mSitkovets for code

**Pull request merged**